src/utils/common.js

```diff
@@ -10,3 +11,27 @@ export async function randomHex (nBytes = 16) {
 	const bytes = await random(nBytes)
 	return bytes.toString('hex')
 }
+
+export async function createAuthUrl(schoolUrl) {
+	const filteredName = schoolUrl.replace('https://', '')
+	const authUrl = 'https://accounts.magister.net/connect/authorize'
+	const authData = {
+		client_id: `M6-${filteredName}`,
+		redirect_uri: `https://${filteredName}/oidc/redirect_callback.html`,
+		response_type: 'id_token token',
+		acr_values: `tenant:${filteredName}`,
+		scope: 'openid profile magister.ecs.legacy magister.mdv.broker.read magister.dnn.roles.read',
+		state: await randomHex(),
+		nonce: await randomHex(),
+	}
+
+	return `${authUrl}?${encodeQuery(authData)}`
+}
+
+export function extractQueryParameter(url, paramater) {
+	if (url.includes('?')) {
+		url = url.split('?')[1]
```

Comment:
[the `url` package](https://nodejs.org/api/url.html#url_urlobject_query) might be more robust.

Reply:
I think the whole use of querystrings can be discarded. The URL package should be able to do everything querystrings is used for here.